File: chaeryeon823/PGS_70129.js
// 이진 변환 반복하기
// lv 2
// 문자열, 구현

function solution(s) {
  var answer = [0, 0];
  while (s !== '1') {
    // 0 제거
    answer[1] += s.split('0').length - 1;
    s = s.replaceAll('0', '');

    // 길이를 2진수
    s = s.length.toString(2);

    answer[0]++;
  }
  console.log(s);
  return answer;
}

File: ahma0/PGS_70129.java
import java.util.Scanner;

class Solution {
    public int[] solution(String s) {
        int cnt = 0;
        int index = 0;

        while (!s.equals("1")) {
            int zero_count = 0;

            for (int i = 0; i < s.length(); i++) {
                if (s.charAt(i) == '0') {
                    zero_count++;
                }
            }

            s = Integer.toBinaryString(s.length() - zero_count);
            cnt += zero_count;
            index++;
        }

        int[] answer = {index, cnt};
        return answer;
    }
}